quiche/src/lib.rs

```diff
@@ -4592,10 +4592,12 @@ impl Connection {
         self.handshake.server_name()
     }
 
-    /// Returns the peer's leaf certificate (if any) as a DER-encoded buffer.
+    /// Returns the certificate at index `i` (if any) in the certificate
+    /// chain exposed by the peer as a DER-encoded buffer. The peer's leaf
+    /// certificate is the one at index 0.
     #[inline]
-    pub fn peer_cert(&self) -> Option<&[u8]> {
-        self.handshake.peer_cert()
+    pub fn peer_cert(&self, i: i32) -> Option<&[u8]> {
```

Comment:
I think it would be better to add a whole new public `peer_cert_chain()` method that returns the full chain in one go, rather than change the existing API. It could return a `Option<Vec<&[u8]>>` or something like that.

Reply:
Thanks for reviewing. Yes, I completely agree, I changed the PR accordingly.